File: C++ DSA/Problems/Rearrange.cpp
#include<iostream>
#include<vector>

using namespace std;

vector<int> reArrange(vector<int>&arr){
    int n = arr.size();
    vector<int>ans(n,0);

    int posIndex = 0,negativeIndex = 1;

    for(int i=0;i<n;i++){
        if(arr[i]<0){
            ans[negativeIndex] = arr[i];
            negativeIndex+=2;
        }else{
            ans[posIndex] = arr[i];
            posIndex+=2;
        }
    }

    return ans;
}

int main(){
    vector<int>inp = {3,1,-2,-5,2,-4};
    vector<int>myResult = reArrange(inp);
    for(int val : myResult){
        cout<<val<<",";
    }
    return 0;
}

File: Stirver SDE Sheet/Arrays & Hashing/Rearrange.js
function reArrange(arr){
    let n = arr.length;

    let positives = [];

    let negative = [];

    for(let i=0;i<n;i++){
        if(arr[i]<0){
            negative.push(arr[i]);
        }else{
            positives.push(arr[i]);
        }
    }

    let result = [];

    for(let i=0;i<positives.length;i++){
       result[i*2] = positives[i];
    }

    for(let i=0;i<negative.length;i++){
        result[i*2+1] = negative[i];
    }

  return result;
}

reArrange([3,1,-2,-5,2,-4]);
reArrange([-1,1]);
reArrange([3,1,-2,-5,2,-4]);